src/hotspot/share/runtime/javaThread.hpp

```diff
@@ -1141,7 +1141,8 @@ class JavaThread: public Thread {
   // java.lang.Thread.sleep support
   ParkEvent * _SleepEvent;
 public:
-  bool sleep(jlong millis);
+  bool sleep_millis(jlong millis);
+  bool sleep_nanos(jlong nanos);
```

Comment:
I prefer just one sleep that takes nanos.

Reply:
If we do only `sleep(jlong nanos)`, then there is an accident waiting to happen, when some unfixed code would call `sleep` with `millis` argument, not knowing it is now `nanos`. That was the reason why I made the names explicit. `sleep_millis` also does the conversion to nanos that does not overflow. But, like with `park` above, I think there is an argument to keep `sleep(millis)` and add `sleep_nanos(nanos)`, to keep code changes at minimum.